### PR TITLE
Add fusion to ptlsbench and fix bcrypt

### PR DESCRIPTION
**Pull request by huitema**
Add the "fusion" implementations to the "ptlsbench" benchmark. These implementations are shown to be slightly faster than those in openssl or on Windows bcrypt.  This does not yet test the "combined packet and header protection" option of the fusion code.

The update required switching from the old ways of calling "init; update; final" to the combined "encrypt" API -- since the partial updates are not supported by the fusion code. The switch surfaced the difference of assumptions between the bcrypt functions and the other AEAD implementations: in bcrypt, the "update" calls only processes 16 bytes block, and squirrels the overflow into the context; the call to "final" must use the "encrypted length so far", which may well be lower than the clear-text length if that length is not a multiple of 16 bytes.

### CODE REVIEW

File: lib/ptlsbcrypt.c [hunk 1]
```diff
@@ -427,6 +427,22 @@ static size_t ptls_bcrypt_aead_do_encrypt_final(struct st_ptls_aead_context_t *_
...
 }
 
+void ptls_bcrypt_do_encrypt(ptls_aead_context_t *ctx, void *output, const void *input, size_t inlen, uint64_t seq,
+                                  const void *aad, size_t aadlen, ptls_aead_supplementary_encryption_t *supp)
+{
+    size_t after_update;
+
+    ctx->do_encrypt_init(ctx, seq, aad, aadlen);
+    after_update = ctx->do_encrypt_update(ctx, output, input, inlen);
+    ctx->do_encrypt_final(ctx, (uint8_t *)output + after_update);
```
Comment (kazuho) [inline lib/ptlsbcrypt.c:437]: Do we want to move this code into `ptls_aead__do_encrypt`? I'd assume that other backends would be fine with this.

**Review by kazuho — COMMENTED** [review]
Thank you for the PR. Looks good, just one question.

File: lib/ptlsbcrypt.c (hunk 1 above)
Comment (huitema) [reply lib/ptlsbcrypt.c:437]: I tried, and I got a FIXME assert when running in release mode in Windows. I think this was caused by a buffer overflow when running the minicrypto version of chachapoly, but I could not easily debug that. So I did a localized change in the bcrypt code instead.

Comment (huitema) [reply lib/ptlsbcrypt.c:437]: @kazuho I would like to merge this now, and perhaps open an issue to address modify `ptls_aead__do_encrypt` later. With the PR as it stands, `ptlsbench` runs fine for all supported algorithms. If I update `ptls_aead__do_encrypt` I know that I will trigger a bug, and I am concerned that debugging will take a fair bit of time.

Comment (kazuho) [reply lib/ptlsbcrypt.c:437]: Thank you. That works for me.

**Pull request merged**